src/NHibernate/AdoNet/OracleDataClientBatchingBatcher.cs

```diff
@@ -125,21 +125,32 @@ protected override void DoExecuteBatch(IDbCommand ps)
 				// this value is not a part of the ADO.NET API.
 				// It's and ODP implementation, so it is being set by reflection
 				SetArrayBindCount(arraySize);
-				int rowsAffected;
 				try
 				{
-					rowsAffected = _currentBatch.ExecuteNonQuery();
+					int rowsAffected;
+					try
+					{
+						rowsAffected = _currentBatch.ExecuteNonQuery();
+					}
+					catch (DbException e)
+					{
+						// Message from DbException could be useful
+						var additionalMessage = string.IsNullOrEmpty(e.Message) ? "." : string.Concat(": ", e.Message);
```

Comment:
Not sure why this is required. I've not run it, but looking at the code is seems the exception ``e`` should eventually end up as inner exception inside the "converted" exception and the message can and should be read from that.